client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -215,8 +217,14 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 			? isThemePurchased( state, selectedDesignThemeId, site.ID )
 			: false
 	);
+
+	const isPluginBundleEligible = useIsPluginBundleEligible();
+	const themeDetails = useThemeDetails( selectedDesign?.slug || '' );
+	const hasThemeSoftwareSet = themeDetails?.data?.taxonomies?.theme_software_set?.length;
+
 	const shouldUpgrade =
-		selectedDesign?.is_premium && ! isPremiumThemeAvailable && ! didPurchaseSelectedTheme;
+		( selectedDesign?.is_premium && ! isPremiumThemeAvailable && ! didPurchaseSelectedTheme ) ||
+		( isEnabled( 'themes/plugin-bundling' ) && ! isPluginBundleEligible && hasThemeSoftwareSet );
```

Comment:
BTW I don't understand why we need to add this clause. Isn't the Thriving Artist theme always a premium theme, so that the existing `shouldUpgrade` value will be already `true`? (So the upgrade modal will be already shown even before we added this new logic)

Reply:
Nevermind, I understand now! 😄